docs/guide/connecting.rst

```diff
@@ -7,30 +7,97 @@
 Connecting to Verilated Models
 ******************************
 
+Structure of the Verilated Model
+================================
+
+Verilator outputs a :file:`{prefix}.h` header file which defines a class
+named :code:`{prefix}` which represents the generated model the user is
+supposed to instantiate.  This Model Class defines the interface of the
+Verilated model.
+
+Verilator will additionally create a :file:`{prefix}.cpp` file, together
+with additional .h and .cpp files for internals.  See the :file:`examples`
+directory in the kit for examples.  See :ref:`Files Read/Written` for
+information on all the files Verilator might output.
+
+The output of Verilator will contain a :file:`{prefix}.mk` file that may be
+used with Make to build a :file:`{prefix}__ALL.a` library with all required
+objects in it.
+
+The generated Model Class file manages all internal state required by the
+model, and exposes the following interface that allows interaction with the
+model:
+
+
+* Top level IO ports are exposed as references to the appropriate internal
+  equivalents.
+
+* Public top level module instances are exposed as pointers to allow access
+  to :code:`/* verilator public */` items.
+
+* The root of the design hierarchy (as in SystemVerilog :code:`$root`) is
+  exposed via the :code:`rootp` member pointer to allow access to model
+  internals, including :code:`/* verilator public_flat */` items.
+
+
+.. _Porting Post 4.204:
+
+Model interface until version 4.204
+-----------------------------------
+
+Up until Verilator version 4.204 inclusive, the generated Model Class was
+also the instance of the top level instance in the design hierarchy (what
+you would refer to with :code:`$root` in SystemVerilog).  This meant that
+all internal variables that were implemented by Verilator in the root scope
+were accessible as members of the Model class itself.  Note there were often
+many such variable due to module inlining, including :code:`/* verilator
+public_flat */` items.  After version 4.204, the Model Class is no longer
+the same as the top level instance, but is an interface object instead.
+This means that user code that accesses internal signals in the model
+(likely including :code:`/* verilator public_flat */` signals) will need to
+be updated as follows:
```

Comment:
Describe how it is now first, as clearer to read when in the future most people understand the present not the past (IMO).

```suggestion
After version 4.204, the Model Class is an interface object.

Up until Verilator version 4.204 inclusive, the generated Model Class was
also the instance of the top level instance in the design hierarchy (what
you would refer to with :code:`$root` in SystemVerilog).  This meant that
all internal variables that were implemented by Verilator in the root scope
were accessible as members of the Model class itself.  Note there were often
many such variable due to module inlining, including :code:`/* verilator
public_flat */` items.

This means that user code that accesses internal signals in the model
(likely including :code:`/* verilator public_flat */` signals) will need to
be updated as follows:
```

Reply:
This is good advice thank you!